### DISCUSSION TITLE
db error: ERROR: incorrect binary data format in bind parameter 3

### DISCUSSION THREAD

**Discussion by muhammad-Osman** [prisma/prisma#16307] [Q&A] [answered]
i am querying the database with the date 

```
const newToRoleResult: any = await prisma.$queryRaw`
        SELECT distinct on (p.id) p.id, count(*), p.start_date, a.grade, a.reviewed_at 
      FROM assessments a, people p WHERE (a.status = CAST('done' AS assessment_status) 
      AND a.reviewee_id = p.id AND 
      (p.institute_id = ${instituteId} 
        OR (p.institute_id in 
          (SELECT inst.id FROM institute inst WHERE inst.mat_id = ${instituteId}))))
          ${
            newToRole == "NTR"
              ? Prisma.sql`AND p.start_date => ${startDate} AND p.start_date =< ${endDate}` : Prisma.empty
          } 
          GROUP BY a.grade, p.id, a.reviewed_at ORDER BY p.id, a.reviewed_at DESC`;
```
but it is always returning me 

db error: ERROR: incorrect binary data format in bind parameter 3

**Comment by nurul3101** [comment]
Hello @muhammad-Osman 👋 

Does this raw query work if you directly execute it in a Database IDE?
What version of Prisma are you using?
Which database are you using?

**Reply by muhammad-Osman** [reply]
Yes it is working on PGADMIN and on terminal as well not working in prisma

OS: Ubuntu
Node Version: v16.13.2
Prisma Version: 3.15.2
Database: Postgresql

**Reply by nurul3101** [reply]
Can you try adding the schema name in front of the table name, for example instead of `assessments`, can you try "public"."assessments", assuming "public" is the schema name.

**Reply by muhammad-Osman** [reply]
> Can you try adding the schema name in front of the table name, for example instead of `assessments`, can you try "public"."assessments", assuming "public" is the schema name.

i did resolved the issue

**Comment by miguelff** [comment, marked as answer]
@muhammad-Osman Could you please provide a schema file?

**Reply by muhammad-Osman** [reply]
> @muhammad-Osman Could you please provide a schema file?

i did resolve this issue 

upgrade to "prisma": "^4.6.0",

then i was facing issue with casting 

```
const newToRoleResult: any = await prisma.$queryRaw`
        SELECT distinct on (p.id) p.id, count(*)::INTEGER, p.start_date, a.grade, a.reviewed_at 
      FROM assessments a, people p WHERE (a.status = CAST('done' AS assessment_status) 
      ${
        newToRole == "NTR"
        ? newToRole
          ? Prisma.sql`AND p.start_date BETWEEN SYMMETRIC ${startDate}::DATE AND ${endDate}::DATE`
          : Prisma.empty
        : newToRole == "NNTR"
        ? Prisma.sql`AND p.start_date NOT BETWEEN SYMMETRIC ${startDate}::DATE AND ${endDate}::DATE`
        : Prisma.empty
      }
      AND a.reviewee_id::text = p.id::text AND 
      (p.institute_id::text = ${instituteId}::text 
        OR (p.institute_id::text in 
          (SELECT inst.id::text FROM institute inst WHERE inst.mat_id::text = ${instituteId}::text))))  
          GROUP BY a.grade, p.id, a.reviewed_at ORDER BY p.id, a.reviewed_at DESC`
```

**Reply by nurul3101** [reply]
Thanks for getting back and letting us know! 🚀

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
